processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -365,7 +365,10 @@ else if ( method.isEnumMapping() ) {
                     .build();
 
                 if ( beanMappingMethod != null ) {
-                    hasFactoryMethod = beanMappingMethod.getFactoryMethod() != null;
+                    // We can consider that the mapping method has a factory method if it has one,
+                    // or if the result type of the methods is not an interface
+                    hasFactoryMethod = beanMappingMethod.getFactoryMethod() != null ||
+                        !beanMappingMethod.getResultType().isInterface();
```

Comment:
As far as I'm concerned that would be a separate issue, right? Possibly there's a more general scenario in that case.